src/notification/notification.cpp

```diff
@@ -66,3 +70,46 @@ bool Notification::push(const QString& message, Severity severity, bool showDial
 
     return true;
 }
+
+bool Notification::clear(void) {
+    if (textBrowser == nullptr) {
+        return false;
+    }
+
+    textBrowser->clear();
+    return true;
+}
+
+bool Notification::saveLog(void) {
+    if (textBrowser == nullptr) {
+        return false;
+    }
+
+    QString fileName = QFileDialog::getSaveFileName(
+        textBrowser->parentWidget(), "", "Log",
+        "Text File (*.txt)\n Markdown File (*.md)\nHTML File (*.html)");
+    std::ofstream file(fileName.toStdString());
+
+    if (!file.is_open()) {
+        push("Log save canceled");
+        return false;
+    }
+
+    QString fileContent;
+    QTextDocument* textDocument = textBrowser->document();
+    std::string fileExtension = std::filesystem::path(fileName.toStdString()).extension().string();
+
+    if (fileExtension == ".html") {
+        fileContent = textDocument->toHtml();
+    } else if (fileExtension == ".md") {
+        fileContent = textDocument->toMarkdown();
```

Comment:
`fileContent = textDocument->toMarkdown().replace("\n\n", "  \n");`
would remove the additional newline in the markdown output. At the same time the double whitespace ensure proper rendering within a markdown engine.